### ISSUE TITLE
Old package with new function causes function to refer to old env

### ISSUE THREAD

**Issue by vishal-biyani** [fission/fission#946]
Create a package with an environment

```
$fission package create --env python --deploy "multifile/*"
Package 'python-1540268954-9jdn' created
```

Now create a new environment and a function which wants to use this package:

```
$ fission env create --name pythonb --image fission/python-env
environment 'pythonb' created
```

```
$ fission fn create --name py98 --env pythonb --pkg python-1540268954-9jdn --entrypoint main.main --executortype newdeploy --minscale 1
function 'py98' created

$ fission fn list
NAME UID                                  ENV    EXECUTORTYPE MINSCALE MAXSCALE MINCPU MAXCPU MINMEMORY MAXMEMORY TARGETCPU
py98 8a25984c-d685-11e8-9b4e-42010aa0001c python newdeploy    1        1        0      0      0         0         80

```

The function still refers to environment `python` because package refers to old environment. Ideally the function should point to `pythonb` environment.

**Comment by life1347** [comment]
two stages for this issue as we discussed
1. Show warning message if the function's env is different package's.
2. Deprecate/Remove the environment reference from function CRD
    - Always create a package for function and assign env reference to it.

**Comment by vishal-biyani** [comment]
Closed by #996
